nx/lib/nx/shape.ex

```diff
@@ -1218,6 +1218,85 @@ defmodule Nx.Shape do
     {List.to_tuple(result_shape), result_names}
   end
 
+  @doc """
```

Comment:
I'm not sure if this file is the best place for this function. Also, name suggestions are more than welcome :)